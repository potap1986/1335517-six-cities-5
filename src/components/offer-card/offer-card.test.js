import OfferCard from './offer-card';
import renderer from 'react-test-renderer';

jest.mock(`../bookmark/bookmark`, () => `Bookmark`);

const offer = {
  hotelCity: {
    name: `Lafayette`,
    location: {
      lat: 30,
      lng: 19,
      zoom: 16,
    },
  },
  previewImage: `mock`,
  images: [`mock`, `mock`],
  title: `mock`,
  isFavorite: true,
  isPremium: true,
  rating: 5,
  type: `hotel`,
  bedrooms: 2,
  adults: 2,
  price: 22,
  goods: [`mock`, `mock`],
  hotelHost: {
    id: 2,
    name: `John Doe`,
    isPro: true,
    avatarUrl: `mock`,
  },
  description: `Mock. Mock.`,
  coordinates: {
    lat: 50,
    lng: 50,
    zoom: 16,
  },
  id: 1,
};

jest.mock(`react-router-dom`, () => ({
  Link: () => null
}));

it(`OfferCard correctly renders after relaunch`, () => {
  const card = renderer
    .create(<OfferCard
      offer={offer}
      onOfferHover={jest.fn()}
      onOfferClick={jest.fn()}
      onBookmarkClick={jest.fn()}
      className={`main`}
    />)
  .toJSON();

  expect(card).toMatchSnapshot();
});
