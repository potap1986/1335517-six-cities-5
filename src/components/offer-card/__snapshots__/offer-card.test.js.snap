// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard correctly renders after relaunch 1`] = `
<article
  className="place-card main__card"
  onMouseOver={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="place-card__image-wrapper main__image-wrapper"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="mock"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info "
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          22
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <Bookmark
        className="place-card__bookmark"
        id={1}
        isFavorite={true}
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        mock
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      hotel
    </p>
  </div>
</article>
`;
